src/main/java/seedu/address/model/entity/IdentificationNumber.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model.entity;
+
+//@@author shaoyi1997
+
+import java.util.Objects;
+
+/**
+ * Represents the ID number of each entity.
+ * Guarantees: immutable, unique ID number will be generated from within the class
+ */
+public class IdentificationNumber {
+
+    private static final String ID_PREFIX_BODY = "B";
+    private static final String ID_PREFIX_WORKER = "W";
+    private static final String ID_PREFIX_FRIDGE = "F";
+
+    private static int countOfBodies = 0;
+    private static int countOfWorkers = 0;
+    private static int countOfFridges = 0;
+
+    private int idNum;
+    private String typeOfEntity;
+
+    // todo: check for duplicates
+    protected IdentificationNumber(String typeOfEntity) {
+        this.typeOfEntity = typeOfEntity;
+        switch (typeOfEntity) {
+        case ID_PREFIX_BODY:
+            countOfBodies++;
+            idNum = countOfBodies;
+            break;
+        case ID_PREFIX_WORKER:
+            countOfWorkers++;
+            idNum = countOfWorkers;
+            break;
+        case ID_PREFIX_FRIDGE:
+            countOfFridges++;
+            idNum = countOfFridges;
+            break;
+        default:
+            throw new IllegalArgumentException("Invalid entity type");
+        }
+    }
+
+    private IdentificationNumber(String typeOfEntity, int idNum) {
+        this.typeOfEntity = typeOfEntity;
+        this.idNum = idNum;
+    }
+
+    public static IdentificationNumber generateNewBodyId() {
+        return new IdentificationNumber(ID_PREFIX_BODY);
+    }
+
+    public static IdentificationNumber generateNewWorkerId() {
+        return new IdentificationNumber(ID_PREFIX_WORKER);
+    }
+
+    public static IdentificationNumber generateNewFridgeId() {
+        return new IdentificationNumber(ID_PREFIX_FRIDGE);
+    }
+
+    public static IdentificationNumber customGenerateId(String typeOfEntity, int idNum) {
```

Comment:
if this is only used for testing, is it possible to package it into test instead of main?